src/graph/impl/KokkosGraph_GraphColor_impl.hpp

```diff
@@ -2379,6 +2379,300 @@ class GraphColor_VB:public GraphColor <HandleType,lno_row_view_t_,lno_nnz_view_t
+
+                               size_type frontierNode = frontier(frontierIdx);
+                               // Initialize bit array to all bits = 0
+                               unsigned long long bannedColors = 0;
+                               color_t myColor = 0, colorOffset = 0;
+
+                               while(myColor == 0) {
+                                 // Loop over neighbors, find banned colors in the range:
+                                 // [colorOffset + 1, colorOffset + 64]
+                                 for(size_type neigh = myXadj(frontierNode); neigh < myXadj(frontierNode + 1); ++neigh) {
+                                   color_t neighColor = colors(myAdj(neigh));
+                                   // Check that the color is in the current range
+                                   if(neighColor > colorOffset && neighColor < colorOffset + 65) {
+                                     // Set bannedColors' bit in location colors(myAdj(neigh)) to 1.
+                                     bannedColors |= (1ULL << (neighColor - 1));
+                                   }
+
+                                   // We want to avoid the cost of atomic operations when not needed
+                                   // so let's check that the node is not already colored, i.e.
+                                   // its dependency is not -1.
+                                   if(colorOffset == 0 && dependency(myAdj(neigh)) >= 0) {
+                                     nnz_lno_t myDependency =
+                                       Kokkos::atomic_fetch_add(&dependency(myAdj(neigh)), -1);
+                                     if(myDependency - 1 == 0) {
+                                       const size_type newFrontierIdx
+                                         = Kokkos::atomic_fetch_add(&newFrontierSize(), 1);
+                                       newFrontier(newFrontierIdx) = myAdj(neigh);
+                                     }
+                                   }
+                                 } // Loop over neighbors
+
+                                 if(~bannedColors == 0ULL) {
+                                   colorOffset += 64;
+                                   // Reset bannedColors to all 0 bits
+                                   bannedColors |= ~bannedColors;
+                                 } else {
+                                   color_t colorIdx = 1;
+                                   // Check if index colordIdx - 1, is set to one in bannedColors
+                                   while(bannedColors & (1ULL << (colorIdx - 1))) {++colorIdx;}
+                                   myColor = colorOffset + colorIdx;
+                                 }
+                               }
+                               colors(frontierNode) = myColor;
+                             }); // Loop over current frontier
+      }
+      Kokkos::deep_copy(host_newFrontierSize, newFrontierSize);
```

Comment:
Is there a fence missing here before ? How are we sure newFrontierSize is updated ?